src/main/java/org/terasology/ligthandshadow/componentsystem/events/MoveFlagToBaseEvent.java

```diff
@@ -0,0 +1,27 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+package org.terasology.ligthandshadow.componentsystem.events;
+
+import org.terasology.engine.entitySystem.entity.EntityRef;
+import org.terasology.engine.entitySystem.event.Event;
+
+/**
+ * This trigger event is responsible for sending the flag back to the base.
+ */
+public class MoveFlagToBaseEvent implements Event {
+    private final EntityRef heldFlag;
+    private final String flagTeam;
```

Comment:
I would call this event `ReturnFlagEvent` and omit the `flagTeam` property, as it is implicit that the flag should be returned to its home base.